### PR TITLE
On riscv64, branch long can using only 2 instr:auipc+jalr.

### PR DESCRIPTION
**Pull request by luyahan**
On riscv64, branch long only using 2 instr auipc+jalr. Refer to [The RISC-V Instruction Set Manual](https://riscv.org/specifications/)

### CODE REVIEW

File: src/codegen/riscv64/macro-assembler-riscv64.cc [hunk 1]
```diff
@@ -3182,10 +3182,10 @@ void TurboAssembler::BranchLong(Label* L) {
...
     imm64 = branch_long_offset(L);
     DCHECK(is_int32(imm64));
-    auipc(t5, 0);  // Read PC into t5.
-    RV_li(t6, imm64);
-    add(t6, t5, t6);
-    jr(t6);
+    int32_t Hi20 = (((int32_t)imm64 + 0x800) >> 12);
+    int32_t Lo12 = (int32_t)imm64 << 20 >> 20;
+    auipc(t5, Hi20);  // Read PC + Hi20 into t5.
+    jalr(t6, Lo12);   // jump PC + Hi20 + Lo12
```
Comment (amyzxf) [inline src/codegen/riscv64/macro-assembler-riscv64.cc:3188]: Is the register in "jalr(t6, Lo12)" right? should it be t5?

File: src/codegen/riscv64/assembler-riscv64.h [hunk 2]
```diff
@@ -641,6 +641,7 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   void jal(int32_t imm21) { jal(ToRegister(1), imm21); }
   inline void jal(Label* L) { jal(jump_offset(L)); }
   void jr(Register rs) { jalr(zero_reg, rs, 0); }
+  void jalr(Register rs, int32_t imm12) { jalr(zero_reg, rs, imm12); }
```
Comment (amyzxf) [inline src/codegen/riscv64/assembler-riscv64.h:644]: Jalr means Jump and Link Register, but the implementation use zero_reg, and the behavior is just jump.  Change the function name "jalr"  to "jr"?

File: src/codegen/riscv64/macro-assembler-riscv64.cc (hunk 1 above)
Comment (luyahan) [reply src/codegen/riscv64/macro-assembler-riscv64.cc:3188]: yeah ,i fix it

File: src/codegen/riscv64/assembler-riscv64.h (hunk 2 above)
Comment (luyahan) [reply src/codegen/riscv64/assembler-riscv64.h]: yes, i fix it now.

**Pull request merged**